src/app/_actions/registerAction.tsx

```diff
@@ -0,0 +1,18 @@
+"use server";
+
+import { registerBodySchema } from "../_types/schemas";
+
+async function registerServerAction(state: any, formData: FormData) {
+  const registerBody = registerBodySchema.safeParse({
+    email: formData.get("email") as string,
+    username: formData.get("username") as string,
+    password: formData.get("password") as string,
+  });
```

Comment:
`formData.get()` returns a weird type, so we have to cast it to a string using `as string`